### PR TITLE
Tab Top Rated Implementation

### PR DESCRIPTION
**Pull request by marcosvonstremayr**
Developed the Top Rated tab which includes the view, bloc, usecase, repository, database and api call. Also added new tests and modified the existing ones. 

https://user-images.githubusercontent.com/109096201/196301692-53bb3ebe-7280-4156-94a9-bc500ea22de2.mp4

No internet connection

https://user-images.githubusercontent.com/109096201/196709034-9fd1a344-aa02-48cb-a1c1-a1137a140d90.mp4

image

### CODE REVIEW

File: lib/data/repository/movies_repository_impl.dart [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  @override
+  Future<MovieEvent> fetchMovies(endpoint) async {
+    List<MovieModel> moviesList = [];
+    int startIndex = endpoint.lastIndexOf('/');
+    String document = endpoint.substring(startIndex + 1) + '_doc';
+    String subcollection = endpoint.substring(startIndex + 1);
+    ConnectivityResult connectivityResult =
+        await _connectivity.checkConnectivity();
+    if (connectivityResult == ConnectivityResult.mobile ||
+        connectivityResult == ConnectivityResult.wifi) {
```
Comment (FranciscoGaraventa) [inline lib/data/repository/movies_repository_impl.dart]: I think you can create a bloc to check connectivity

File: lib/main.dart [hunk 2]
```diff
@@ -1,18 +1,67 @@
...
+  late IMoviesBloc _moviesBloc;
+  late Connectivity _connectivity;
+
+  @override
+  void initState() {
+    _client = http.Client();
+    _db = Database();
+    _service = ApiService(client: _client);
+    _connectivity = Connectivity();
+    _moviesRepository = MoviesRepository(_service, _db, _connectivity);
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: trailing comma

File: lib/data/repository/movies_repository_impl.dart (hunk 1 above)
Comment (marcosvonstremayr) [reply lib/data/repository/movies_repository_impl.dart]: done

File: lib/main.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]

**Review by CIFUM007 — APPROVED** [review]

**Review by MarcosCGlobant — APPROVED** [review]

**Pull request merged**